# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/browser-fullZoom.js
# Commit: 4230bdcf07ae
# Full Hash: 4230bdcf07ae69e03a9e241f10c201c38e6ca995
# Author: Mike Conley <mconley@mozilla.com>
# Date: 2022-06-23 03:54:50
# Regressor Bug: 1773865
# File Overlap Count: 1
# Description:
#   Bug 1773865 - Dispatch an event on the window document when a pinch zoom gesture ends. r=botond,smaug,NeilDeakin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D149283
# ==============================================================================

diff -r d1ddb1ce916b -r 4230bdcf07ae browser/base/content/browser-fullZoom.js
--- a/browser/base/content/browser-fullZoom.js	Wed Jun 22 16:08:22 2022 +0000
+++ b/browser/base/content/browser-fullZoom.js	Wed Jun 22 16:15:57 2022 +0000
@@ -50,6 +50,7 @@
   init: function FullZoom_init() {
     gBrowser.addEventListener("DoZoomEnlargeBy10", this);
     gBrowser.addEventListener("DoZoomReduceBy10", this);
+    window.addEventListener("MozScaleGestureComplete", this);
 
     // Register ourselves with the service so we know when our pref changes.
     this._cps2 = Cc["@mozilla.org/content-pref/service;1"].getService(
@@ -86,6 +87,7 @@
     this._cps2.removeObserverForName(this.name, this);
     gBrowser.removeEventListener("DoZoomEnlargeBy10", this);
     gBrowser.removeEventListener("DoZoomReduceBy10", this);
+    window.removeEventListener("MozScaleGestureComplete", this);
   },
 
   // Event Handlers
@@ -100,6 +102,11 @@
       case "DoZoomReduceBy10":
         this.changeZoomBy(this._getTargetedBrowser(event), -0.1);
         break;
+      case "MozScaleGestureComplete": {
+        let nonDefaultScalingZoom = event.detail != 1.0;
+        this.updateCommands(nonDefaultScalingZoom);
+        break;
+      }
     }
   },
 
@@ -323,7 +330,18 @@
 
   // update state of zoom menu items
 
-  updateCommands: function FullZoom_updateCommands() {
+  /**
+   * Updates the current windows Zoom commands for zooming in, zooming out
+   * and resetting the zoom level. Dispatches a ZoomCommandsUpdated event on
+   * the window when the commands have been updated.
+   *
+   * @param {boolean} [forceResetEnabled=false]
+   *   Set to true if the zoom reset command should be enabled regardless of
+   *   whether or not the ZoomManager.zoom level is at 1.0. This is specifically
+   *   for when using scaling zoom via the pinch gesture which doesn't cause
+   *   the ZoomManager.zoom level to change.
+   */
+  updateCommands: function FullZoom_updateCommands(forceResetEnabled = false) {
     let zoomLevel = ZoomManager.zoom;
     let reduceCmd = document.getElementById("cmd_fullZoomReduce");
     if (zoomLevel == ZoomManager.MIN) {
@@ -340,7 +358,7 @@
     }
 
     let resetCmd = document.getElementById("cmd_fullZoomReset");
-    if (zoomLevel == 1) {
+    if (zoomLevel == 1 && !forceResetEnabled) {
       resetCmd.setAttribute("disabled", "true");
     } else {
       resetCmd.removeAttribute("disabled");